File: src/main/java/net/dontdrinkandroot/gitki/wicket/event/FileEvent.java
package net.dontdrinkandroot.gitki.wicket.event;

import net.dontdrinkandroot.gitki.model.FilePath;
import org.apache.wicket.ajax.AjaxRequestTarget;

/**
 * @author Philip Washington Sorst <philip@sorst.net>
 */
public class FileEvent extends AbstractEvent
{
    private FilePath filePath;

    public FileEvent(FilePath filePath)
    {
        this.filePath = filePath;
    }

    public FileEvent(FilePath filePath, AjaxRequestTarget target)
    {
        super(target);
        this.filePath = filePath;
    }

    public FilePath getFilePath()
    {
        return this.filePath;
    }
}

File: src/main/kotlin/net/dontdrinkandroot/gitki/wicket/event/FileEvent.kt
package net.dontdrinkandroot.gitki.wicket.event

import net.dontdrinkandroot.gitki.model.FilePath
import org.apache.wicket.ajax.AjaxRequestTarget

open class FileEvent : AbstractEvent {

    var filePath: FilePath
        private set

    constructor(filePath: FilePath) {
        this.filePath = filePath
    }

    constructor(filePath: FilePath, target: AjaxRequestTarget?) : super(target) {
        this.filePath = filePath
    }
}